## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#2477c93f-fd56-4ff1-8423-3f67668e6671) |
| mint-large-bucket.sh | :x: [more...](#c8cae546-6e05-4f15-9a83-1f99dca812e7) |
| mint-fs.sh | :x: [more...](#5cf2066d-60b8-4410-b0e6-2e455a0b7f1b) |
| mint-dist-xl.sh | :x: [more...](#9900367c-3a5a-4446-b2d0-fd1c6a99dedb) |
<a name="c8cae546-6e05-4f15-9a83-1f99dca812e7"></a>
### 8464-f4894ac/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:32511
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5f2c2d9be068:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 0 seconds
(4/14) Running aws-sdk-ruby tests ... done in 14 seconds
(5/14) Running awscli tests ... done in 2 minutes and 24 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 12 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 21 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 7 seconds
{
  "args": {},
  "duration": 104351,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="5cf2066d-60b8-4410-b0e6-2e455a0b7f1b"></a>
### 8464-f4894ac/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31997
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3f861727f948:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 55 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 54 seconds
(8/14) Running minio-dotnet tests ... done in 26 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 8 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 36 seconds
(11/14) Running minio-js tests ... done in 43 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 12 seconds
{
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "message": "AccessDenied: message: Access Denied",
  "args": {
    "presigned_get_object_url": "http://72.28.97.53:31997/minio-py-test-b7b3ffcd-75f0-423c-8cd1-72af5f80c081/b662ebf2-d5ec-4ca3-8560-5d18045de647?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191029%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191029T164908Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=11cfaec76046227cfe01ab317bf2841defed011db65c0220c8541f60f1fbcd10",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-10-29T16-20-05Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15D22C25C8A5C210\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Tue, 29 Oct 2019 16:49:20 GMT\"]}}",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>b662ebf2-d5ec-4ca3-8560-5d18045de647</Key><BucketName>minio-py-test-b7b3ffcd-75f0-423c-8cd1-72af5f80c081</BucketName><Resource>/minio-py-test-b7b3ffcd-75f0-423c-8cd1-72af5f80c081/b662ebf2-d5ec-4ca3-8560-5d18045de647</Resource><RequestId>15D22C25C8A5C210</RequestId><HostId>43182913-e6df-487f-8e64-4b1e724709b6</HostId></Error>",
    "response.reason": "Forbidden",
    "response.status": 403,
    "object_name": "b662ebf2-d5ec-4ca3-8560-5d18045de647",
    "bucket_name": "minio-py-test-b7b3ffcd-75f0-423c-8cd1-72af5f80c081"
  },
  "duration": 58,
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```
<a name="9900367c-3a5a-4446-b2d0-fd1c6a99dedb"></a>
### 8464-f4894ac/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30000
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 906da54f8746:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 1 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 50 seconds
(8/14) Running minio-dotnet tests ... done in 43 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 51 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 13 seconds
(11/14) Running minio-js tests ... done in 1 minutes and 0 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 55 seconds
{
  "message": "AccessDenied: message: Access Denied",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL",
  "args": {
    "response.status": 403,
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>ec0cc2ec-b719-4b96-94af-5c0ea4314afe</Key><BucketName>minio-py-test-0e42a3cc-2c7a-42a1-b09d-10a65ddb7b6c</BucketName><Resource>/minio-py-test-0e42a3cc-2c7a-42a1-b09d-10a65ddb7b6c/ec0cc2ec-b719-4b96-94af-5c0ea4314afe</Resource><RequestId>15D22C4F71EE8A2A</RequestId><HostId>7278f83a-b199-4756-ab1f-f035fba052b1</HostId></Error>",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-10-29T16-20-05Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15D22C4F71EE8A2A\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Tue, 29 Oct 2019 16:52:19 GMT\"]}}",
    "object_name": "ec0cc2ec-b719-4b96-94af-5c0ea4314afe",
    "response.reason": "Forbidden",
    "presigned_get_object_url": "http://72.28.97.51:30000/minio-py-test-0e42a3cc-2c7a-42a1-b09d-10a65ddb7b6c/ec0cc2ec-b719-4b96-94af-5c0ea4314afe?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191029%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191029T165206Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=a4b5b35a86bedb92b3d93d4d5ddf43b080d0334d98684cd5828edd863656fba2",
    "bucket_name": "minio-py-test-0e42a3cc-2c7a-42a1-b09d-10a65ddb7b6c"
  },
  "duration": 113,
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n"
}

Executed 11 out of 14 tests successfully.
```
<a name="2477c93f-fd56-4ff1-8423-3f67668e6671"></a>
### 8464-f4894ac/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:32742
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 83e191f9c3bf:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 52 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 58 seconds
(8/14) Running minio-dotnet tests ... done in 23 seconds
(9/14) Running minio-go tests ... done in 51 seconds
(10/14) Running minio-java tests ... FAILED in 22 seconds
error occurred
ErrorResponse(code=BucketNotEmpty, message=The bucket you tried to delete is not empty, bucketName=minio-java-test-2nb4g19, objectName=null, resource=/minio-java-test-2nb4g19, requestId=15D22BE47D773798, hostId=2a28e0aa-c0be-4533-8a1e-703f57d0178c)
request={method=DELETE, url=http://72.28.97.52:32742/minio-java-test-2nb4g19, headers=Host: 72.28.97.52:32742
User-Agent: MinIO (amd64; amd64) minio-java/dev
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20191029T164440Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20191029/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=409, headers=Accept-Ranges: bytes
Content-Length: 328
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO/DEVELOPMENT.2019-10-29T16-20-05Z
Vary: Origin
X-Amz-Request-Id: 15D22BE47D773798
X-Xss-Protection: 1; mode=block
Date: Tue, 29 Oct 2019 16:44:40 GMT
}

	at io.minio.MinioClient.executeReq(MinioClient.java:1204)
	at io.minio.MinioClient.execute(MinioClient.java:1066)
	at io.minio.MinioClient.executeDelete(MinioClient.java:1372)
	at io.minio.MinioClient.removeBucket(MinioClient.java:3693)
	at FunctionalTest.teardown(FunctionalTest.java:403)
	at FunctionalTest.runTests(FunctionalTest.java:3075)
	at FunctionalTest.main(FunctionalTest.java:3148)

Executed 9 out of 14 tests successfully.
```
Error: No such image: minio/minio:8464-f4894ac

--- a/pkg/s3select/sql/value.go
+++ b/pkg/s3select/sql/value.go
@@ -342,7 +342,7 @@ const (
 
 // InferBytesType will attempt to infer the data type of bytes.
 // Will fail if value type is not bytes or it would result in invalid utf8.
-// ORDER: int, float, bool, JSON, timestamp, string
+// ORDER: int, float, bool, JSON (object or array), timestamp, string
 // If the content is valid JSON, the type will still be bytes.
 func (v *Value) InferBytesType() (err error) {
 	b, ok := v.ToBytes()
@@ -351,29 +351,26 @@ func (v *Value) InferBytesType() (err error) {
 	}
 
 	// Check for numeric inference
-	iA, okAi := v.bytesToInt()
-	if okAi {
-		v.setInt(iA)
+	if x, ok := v.bytesToInt(); ok {
+		v.setInt(x)
+		return nil
+	}
+	if x, ok := v.bytesToFloat(); ok {
+		v.setFloat(x)
+		return nil
+	}
+	if x, ok := v.bytesToBool(); ok {
+		v.setBool(x)
 		return nil
 	}
 
-	fA, okAf := v.bytesToFloat()
-	if okAf {
-		v.setFloat(fA)
+	asString := strings.TrimSpace(v.bytesToString())
+	if len(b) > 0 &&
+		(strings.HasPrefix(asString, "{") || strings.HasPrefix(asString, "[")) {
 		return nil
 	}
 
-	bA, okAb := v.bytesToBool()
-	if okAb {
-		v.setBool(bA)
-		return nil
-	}
-
-	if json.Valid(b) {
-		return nil
-	}
-
-	if t, err := parseSQLTimestamp(v.bytesToString()); err == nil {
+	if t, err := parseSQLTimestamp(asString); err == nil {
 		v.setTimestamp(t)
 		return nil
 	}
@@ -381,8 +378,7 @@ func (v *Value) InferBytesType() (err error) {
 		return errors.New("value is not valid utf-8")
 	}
 	// Fallback to string
-	sA := v.bytesToString()
-	v.setString(sA)
+	v.setString(asString)
 	return
 }
 

